### PR TITLE
Integrate source performance harness with datadog

### PR DESCRIPTION
**Pull request by xiaohansong**
## What
https://github.com/airbytehq/airbyte/issues/31363

Integrate performance harness with datadog. This will send out a metric report after the run to datadog and datadog could collect them as a metric. Example: 

https://app.datadoghq.com/metric/explorer?start=1697235297143&end=1697238897143&paused=false#N4Ig7glgJg5gpgFxALlAGwIYE8D2BXJVEADxQEYAaELcqyKBAC1pEbghkcLIF8qo4AMwgA7CAgg4RKUAiwAHOChASAtnADOcAE4RNIKtrgBHPJoQaUAbVBGN8qVoD6gnNtUZCKiOq279VKY6epbINiAiGOrKQdpYZAYgUJ4YThr42gDGSsgg6gi6mZaBZnHKGABuMMiZUiJwmQhuGgB0itqu7hgi2S1M2vic8gTAAFQ8IDwAulSdqniYoeFzCxgxpfGTMyAacmg5oPIY+wgIOUk4ME6ZixoQmYloonBOcorK6U9Qj89O9EzKERuDxoSb8CD2TBYV4Kc5PepbHh8HbyJ4IADCUmEMBQIgWaB4QA

## How
Use Datadog Client to send a HTTP request with corresponding DD_API_KEY, which is stored as a secret in airbyte. I asked @git-phu to enable that secret for this repo so we would be able to reuse that.

Change k8s env and related command in GHA workflow so we can pass along these env variables.

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/resources/pull-requests-handbook)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:generateScaffolds` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

<details><summary><strong>Updating the Python CDK</strong></summary>

### Airbyter

Before merging:
- Pull Request description explains what problem it is solving
- Code change is unit tested
- Build and my-py check pass
- Smoke test the change on at least one affected connector
   - On Github: Run [this workflow](https://github.com/airbytehq/airbyte/actions/workflows/connectors_tests.yml), passing `--use-local-cdk --name=source-<connector>` as options
   - Locally: `airbyte-ci connectors --use-local-cdk --name=source-<connector> test`
- PR is reviewed and approved
      
After merging:
- [Publish the CDK](https://github.com/airbytehq/airbyte/actions/workflows/publish-cdk-command-manually.yml)
   - The CDK does not follow proper semantic versioning. Choose minor if this the change has significant user impact or is a breaking change. Choose patch otherwise.
   - Write a thoughtful changelog message so we know what was updated.
- Merge the platform PR that was auto-created for updating the Connector Builder's CDK version
   - This step is optional if the change does not affect the connector builder or declarative connectors.

</details>

### CODE REVIEW

File: .github/workflows/connector-performance-command.yml [hunk 1]
```diff
@@ -180,6 +180,7 @@ jobs:
           export STREAM_NUMBER=$STREAM_NUMBER
           export SYNC_MODE=$SYNC_MODE
           export HARNESS=$HARNESS_TYPE
+          export DD_API_KEY=${{ secrets.DD_API_KEY }}
```
Comment (git-phu) [inline .github/workflows/connector-performance-command.yml:183]: You can set this env var in the `env:` section of this step

File: .github/workflows/connector-performance-command.yml [hunk 2]
```diff
@@ -180,6 +181,7 @@ jobs:
           export STREAM_NUMBER=$STREAM_NUMBER
           export SYNC_MODE=$SYNC_MODE
           export HARNESS=$HARNESS_TYPE
+          export DD_API_KEY=$DD_API_KEY
```
Comment (git-phu) [inline .github/workflows/connector-performance-command.yml:184]: Maybe I'm missing something here, but why do we need to export like this when the env var is already set?

Comment (xiaohansong) [reply .github/workflows/connector-performance-command.yml:184]: I’m following the comment on line 178 - seems we need to export those to be accessible to envsubst- though I haven’t tested if that comment is true in this case

Comment (xiaohansong) [reply .github/workflows/connector-performance-command.yml]: you are right - that comment wasn't true. Fixed and cleaned it up.

Comment (xiaohansong) [reply .github/workflows/connector-performance-command.yml]: and it still works! https://github.com/airbytehq/airbyte/actions/runs/6540161486/job/17759616382 so you are right, we don't need to re-export stuff that has already been set in env.

File: airbyte-integrations/connectors-performance/source-harness/src/main/java/io/airbyte/integrations/source_performance/PerformanceTest.java [hunk 3]
```diff
@@ -105,8 +123,44 @@ void runTest() throws Exception {
...
+    log.info("total secs: {}. total MB read: {}, rps: {}, throughput: {}", totalTimeSecs, totalMB, rps, throughput);
     source.close();
+
+    final long reportingTimeInEpochSeconds = OffsetDateTime.now().toInstant().getEpochSecond();
+
+    List<MetricResource> metricResources = List.of(
+        new MetricResource().name("github").type("runner"),
+        new MetricResource().name(imageName).type("image"),
+        new MetricResource().name(dataset).type("dataset"));
+    MetricPayload body =
```
Comment (rodireich) [inline airbyte-integrations/connectors-performance/source-harness/src/main/java/io/airbyte/integrations/source_performance/PerformanceTest.java:136]: amazing!

**Review by rodireich — APPROVED** [review]

**Pull request merged**